GAWB-1435: tsv entity download now POST and accepts columns as form data

davidangb opened the pull request:
requires broadinstitute/firecloud-orchestration#337

It was possible for the comma-delimited list of column names (plus the base url) to exceed 2048 characters, which caused a spray error.

This PR switches to POST-ing the list of column names via form data.

Still left to do/open questions:
* is it okay to hardcode the id of the form? Is there a better way to reference the form, other than `(.submit (.getElementById js/document "tsvdownloadform"))`?
* the tsv endpoint is still under `/cookie-authed/`, and we no longer use a cookie. Should this move? Attempting to set the cookie onClick and then submit the form wasn't safe - the cookie wouldn't get set in time.

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

davidangb committed:
tsv entity download now POST and accepts columns as form data

dmohs committed:
Use ref instead of element ID

zarsky-broad committed:
using semantic submit input instead of ref

dmohs commented:
* is it okay to hardcode the id of the form? Is there a better way to reference the form, other than `(.submit (.getElementById js/document "tsvdownloadform"))`?

It is not ideal. See my commit for the alternative.

Then see Isaac's commit for a better alternative.

* the tsv endpoint is still under `/cookie-authed/`, and we no longer use a cookie. Should this move? Attempting to set the cookie onClick and then submit the form wasn't safe - the cookie wouldn't get set in time.

I'm suspicious of this claim. I have not been able to reproduce the cookie "not getting set in time" after trying many times to reproduce.

zarsky-broad committed:
Update from develop

Merge conflict in
src/cljs/org/broadinstitute/firecloud_ui/page/workspace/data/tab.cljs
I think I dealt with it correctly.

zarsky-broad committed:
Revert "Update from develop"

This reverts commit 585149855d94b044b32baf513c9e770420c6d6d9.

zarsky-broad committed:
Merge branch 'develop' into da_GAWB-1435_tsvDownloadColumns

# Conflicts:
#	src/cljs/org/broadinstitute/firecloud_ui/page/workspace/data/tab.cljs

zarsky-broad committed:
properly re-applying changes

zarsky-broad committed:
include FCtoken in tsv download post

dmohs reviewed (approved):
Looks good to me.
